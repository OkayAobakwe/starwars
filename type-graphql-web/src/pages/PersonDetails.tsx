import { Box, Stack, Flex, Heading, Text, Center, Button } from "@chakra-ui/react";
import React from "react";
import {  usePersonDetails } from "../context/PersonDetailContext";
import { useHistory } from "react-router-dom";

const PersonDetails: React.FC = () => {
  const history = useHistory();
  const {person} = usePersonDetails();
  
  return (
    <>
      <Button onClick={() => history.push("/")}>Back</Button>
      <Center>
        <Box
          maxW={"445px"}
          w={"full"}
          bg="white"
          boxShadow={"2xl"}
          rounded={"md"}
          p={6}
          overflow={"hidden"}
        >
          <Stack>
            <Flex justifyContent="space-between">
              <Heading
                color="gray.700"
                fontSize={"2xl"}
                fontFamily={"body"}
              >
                {person?.person?.name}
              </Heading>
            </Flex>
            <Text
              color={"green.500"}
              textTransform={"uppercase"}
              fontWeight={800}
              fontSize={"sm"}
              letterSpacing={1.1}
            >
              height: {person?.person?.height}
            </Text>
            <Heading
              color="gray.700"
              fontSize={"2xl"}
              fontFamily={"body"}
            >
              {person?.gender}
            </Heading>
            <Text color={"gray.500"}> mass: {person?.person?.mass}</Text>
          </Stack>
          <Stack mt={6} direction={"row"} spacing={4} align={"center"}>
            <Stack direction={"column"} spacing={0} fontSize={"sm"}>
              <Text 
                color={"yellow.500"}
                textTransform={"uppercase"}
                fontWeight={800}
                fontSize={"sm"}
                letterSpacing={1.1}
              >
                Home World: {person?.person?.homeworld?.name}
              </Text>
              <Text color={"gray.500"}>Population: {person?.person?.homeworld?.population}</Text>
              <Text color={"gray.500"}>Terrain: {person?.person?.homeworld?.terrain}</Text>
            </Stack>
          </Stack>
        </Box>
      </Center>
    </>
  )
}
export default PersonDetails