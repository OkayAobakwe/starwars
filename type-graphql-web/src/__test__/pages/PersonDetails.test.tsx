import  PersonDetails  from "../../pages/PersonDetails";
import { render } from "@testing-library/react";

describe("Person Details", () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = render(<PersonDetails />);
  });

  it("renders successfully", () => {
    expect(wrapper).toMatchSnapshot();
  });
});