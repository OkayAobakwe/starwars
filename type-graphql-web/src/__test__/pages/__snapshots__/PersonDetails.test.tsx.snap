// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Details renders successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <button
        class="chakra-button css-5c2176"
        type="button"
      >
        Back
      </button>
      <div
        class="css-gmuwbf"
      >
        <div
          class="css-16hc176"
        >
          <div
            class="chakra-stack css-n21gh5"
          >
            <div
              class="css-gg4vpm"
            >
              <h2
                class="chakra-heading css-1r8d57x"
              />
            </div>
            <p
              class="chakra-text css-13ub55k"
            >
              height: 
            </p>
            <h2
              class="chakra-heading css-1r8d57x"
            />
            <p
              class="chakra-text css-ax9bdp"
            >
               mass: 
            </p>
          </div>
          <div
            class="chakra-stack css-v0cdwd"
          >
            <div
              class="chakra-stack css-1r2fu7y"
            >
              <p
                class="chakra-text css-11inmed"
              >
                Home World: 
              </p>
              <p
                class="chakra-text css-ax9bdp"
              >
                Population: 
              </p>
              <p
                class="chakra-text css-ax9bdp"
              >
                Terrain: 
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      class="chakra-button css-5c2176"
      type="button"
    >
      Back
    </button>
    <div
      class="css-gmuwbf"
    >
      <div
        class="css-16hc176"
      >
        <div
          class="chakra-stack css-n21gh5"
        >
          <div
            class="css-gg4vpm"
          >
            <h2
              class="chakra-heading css-1r8d57x"
            />
          </div>
          <p
            class="chakra-text css-13ub55k"
          >
            height: 
          </p>
          <h2
            class="chakra-heading css-1r8d57x"
          />
          <p
            class="chakra-text css-ax9bdp"
          >
             mass: 
          </p>
        </div>
        <div
          class="chakra-stack css-v0cdwd"
        >
          <div
            class="chakra-stack css-1r2fu7y"
          >
            <p
              class="chakra-text css-11inmed"
            >
              Home World: 
            </p>
            <p
              class="chakra-text css-ax9bdp"
            >
              Population: 
            </p>
            <p
              class="chakra-text css-ax9bdp"
            >
              Terrain: 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
